File: gfg-weekly-coding-contest-107/Weight Problem.py

class Solution:
    # Version 1: Count 1 in binary representation
    # TC: O(logn), SC: O(logn)
    def nBlocks(self, w : int) -> int:
        # code here
        bw = bin(w)[2:]
        return bw.count('1')
        

File: gfg-weekly-coding-contest-107/Weight Problem.cpp

class Solution {
  public:
    // Version 1: Count 1 in binary representation
    // TC: O(logn), SC: O(logn)
    int nBlocks(int w) {
        // code here
        int ans = 0;
        while (w > 0) {
            ans += (w & 1);
            w >>= 1;
        }
        return ans;
    }
};